libs/core/src/components/pds-button/pds-button.tsx

```diff
@@ -185,9 +223,7 @@ export class PdsButton {
           </span>
         )}
 
-        {this.variant === 'disclosure' &&
-          <pds-icon class={this.loading ? 'pds-button__icon--hidden' : ''} icon={caretDown} part="caret" aria-hidden="true"></pds-icon>
-        }
+        {this.renderEndContent()}
```

Comment:
When there is an icon placed using the slot, when I set the button to loading ,the icon does not disappear. See screenshot.

![Screenshot 2025-04-23 at 2 42 06 PM](https://github.com/user-attachments/assets/9ea70a45-2fa6-4b7d-a259-6e2aea39c6e6)